pkg/action/build-iso.go

```diff
@@ -187,27 +186,30 @@ func (b *BuildISOAction) ISORun() error {
 }
 
 func (b *BuildISOAction) PrepareEFI(rootDir, uefiDir string) error {
-	return b.bootloader.InstallEFIFallbackBinaries(rootDir, uefiDir, b.spec.Label)
+	err := b.renderGrubTemplate(uefiDir)
+	if err != nil {
+		return err
+	}
+	return b.bootloader.InstallEFI(rootDir, uefiDir)
 }
 
 func (b *BuildISOAction) PrepareISO(rootDir, imageDir string) error {
-	err := utils.MkdirAll(b.cfg.Fs, filepath.Join(imageDir, grubPrefixDir), constants.DirPerm)
+	// Include EFI contents in iso root too
+	return b.PrepareEFI(rootDir, imageDir)
```

Comment:
With this commit PrepareEFI and PrepareISO are just identical, both simply prepare EFI bootloader. One is used while booting as an ISO and the other while booting the ISO as a disk (dd ISO to a raw image and boot it as a disk)